File: src/main/java/net/gamershub/kitpvp/events/PlayerSpawnEvent.java
package net.gamershub.kitpvp.events;

import lombok.Getter;
import org.bukkit.entity.Player;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;
import org.jetbrains.annotations.NotNull;

@Getter
public class PlayerSpawnEvent extends Event {
    private static final HandlerList HANDLER_LIST = new HandlerList();

    private final Player player;

    public PlayerSpawnEvent(Player player) {
        this.player = player;
    }

    @Override
    public @NotNull HandlerList getHandlers() {
        return HANDLER_LIST;
    }

    public static HandlerList getHandlerList() {
        return HANDLER_LIST;
    }

}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/events/PlayerSpawnEvent.kt
package io.github.chaosdave34.kitpvp.events

import org.bukkit.entity.Player
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

class PlayerSpawnEvent(val player: Player) : Event() {
    override fun getHandlers(): HandlerList {
        return handlerList
    }

    companion object {
        private val handlerList = HandlerList()

        @JvmStatic
        fun getHandlerList(): HandlerList = handlerList
    }
}